### PR TITLE
Replace '_' with '-' in version prefix

### PR DESCRIPTION
**Pull request by ghost**
@sandersaares Please let me know what do you think about the fix. I believe we also need to update RegexPatterns.cs in the middle-lane repo. Is that correct?

### CODE REVIEW

File: sandersaares.devops-scripts/Set-VersionStringBranchPrefix.ps1 [hunk 1]
```diff
@@ -66,6 +66,8 @@ function Set-VersionStringBranchPrefix {
             # What? Not sure - should have been a reference with the /s in it but okay, just go with it.
             $versionPrefix = $sourceRef
         }
+
+        $versionPrefix = $versionPrefix.Replace("_", "-");
```
Comment (sandersaares) [inline sandersaares.devops-scripts/Set-VersionStringBranchPrefix.ps1]: Would also be nice to leave a comment here explaining why this sort of conversion/sanitization is needed.

**Comment by sandersaares** [comment]
> I believe we also need to update RegexPatterns.cs in the middle-lane repo. Is that correct?

No, that file is intentionally restrictive and needs to stay so because it protects us against violating Azure naming constraints - rather the point right now is that we need to ensure that the version string matches what we have in RegexPatterns.cs. I think this change should be enough - now we just need to flow it through to the right place. Ping me on Teams when you are online and this PR is finalized - we can do the propagation of this change together.

**Comment by sandersaares** [comment]
Note that there are multiple paths to set the branch prefix in this script. For example, line 52 might cause the block on line 58 to not be entered. Let's make sure we cover all the cases.

**Comment by ghost** [comment]
> > I believe we also need to update RegexPatterns.cs in the middle-lane repo. Is that correct?
> 
> No, that file is intentionally restrictive and needs to stay so because it protects us against violating Azure naming constraints - rather the point right now is that we need to ensure that the version string matches what we have in RegexPatterns.cs. I think this change should be enough - now we just need to flow it through to the right place. Ping me on Teams when you are online and this PR is finalized - we can do the propagation of this change together.

Thanks for the explanation!

**Comment by ghost** [comment]
> Note that there are multiple paths to set the branch prefix in this script. For example, line 52 might cause the block on line 58 to not be entered. Let's make sure we cover all the cases.

Thanks for the comment. I've moved the code to cover all the cases.

**Pull request merged**